backend/tournesol/views/inconsistencies.py

```diff
@@ -0,0 +1,216 @@
+                inconsistent_comparisons_count[criterion] += 1
+
+                inconsistent_criterion_comparisons.append(
+                    {
+                        "inconsistency": inconsistency,
+                        "criterion": criterion,
+                        "entity_1_uid": entity_1,
+                        "entity_2_uid": entity_2,
+                        "comparison_score": comparison_score,
+                        "entity_1_rating": rating_1["score"],
+                        "entity_2_rating": rating_2["score"],
+                    }
+                )
+
+        inconsistent_criterion_comparisons.sort(key=lambda d: d['inconsistency'], reverse=True)
+
+        response["count"] = len(inconsistent_criterion_comparisons)
+        response["results"] = inconsistent_criterion_comparisons
+        response["stats"] = {}
+        for criterion in criteria_list:
+            response["stats"][criterion] = {
+                "mean_inconsistency": mean_inconsistency[criterion],
+                "inconsistent_comparisons_count": inconsistent_comparisons_count[criterion],
+                "comparisons_count": comparisons_analysed_count[criterion],
+            }
+
+        return response
+
+    @staticmethod
+    def _calculate_inconsistency(entity_1_calculated_rating,
+                                 entity_2_calculated_rating,
+                                 comparison_score,
+                                 uncertainty) -> float:
+        """
+        Calculate the inconsistency between the comparison
+        criterion score and the general rating of the entity.
+
+        Let's note R the rating difference (rating2 - rating1),
+        r its variable counterpart, U the uncertainty and C the
+        comparison (positive if the 2nd entity is preferred).
+        Consider this function i:
+        i(r) = |C - 10r / sqrt(r² + 1)| (cf issue #671 discussion)
+        The inconsistency is the minimum of i(r) for r in [R - U, R + U]
+
+        The function i(r), is either:
+        1 - always increasing, if c <= -10 (normally, c is in [-10, 10])
+        2 - decreasing to 0, then increasing (because it is an absolute value)
+        3 - always decreasing, if c >= 10
+        So, the inconsistency is either i(R - U) in case 1, inconsistency(R + U)
+        in case 3, or 0 in case 2.
+        The value of R for which i(r) = 0 is C / sqrt(100 - C²), which
+        helps in case 2 to know if the minimum is 0 by checking if this root
+        is in [R - U, R + U].
+
+        There is also an imprecision of 0.5 on the comparisons,
+        since the comparisons are made on integers (considering that contributors
+        have to "mentally round" their preferences to the nearest integer).
+        This imprecision is subtracted on the result of the previous calculation.
+        """
+
+        base_rating_difference = entity_2_calculated_rating - entity_1_calculated_rating
+        comparison_max = 10
```

Comment:
Ok to modify that.
I also think constants.py should be in tournesol/utils, not in core/utils.
Besides, I think the MAX_VALUE for ratings should be renamed RATING_MAX, put in constants.py and have its comment fixed (it is not between 0 and 100, but between -100 and 100). I can change that in this PR if you agree.

Reply:
Besides, the score inconsistencies would be more relevant if, whenever you make a new comparison, the personal and global ratings for this entity are updated immediately. If possible, we should try to enable the machine learning for a single entity, and for a single (user/entity) pair, otherwise we could face numerous small issues.

Reply:
> I think the MAX_VALUE for ratings should be renamed RATING_MAX

This one is never used. I think it is legacy code that instead should be removed.

> I also think constants.py should be in tournesol/utils, not in core/utils. This would be better in a specific PR, as it requires to modify multiple files.

Yes, or we could keep two `constants.py` files in the respective apps with the constants that they use most. But as of now all constants are related to the `tournesol` app

> Besides, the score inconsistencies would be more relevant if, whenever you make a new comparison, the personal and global ratings for this entity were updated immediately. If possible, we should try to enable the machine learning for a single entity, and for a single (user/entity) pair, otherwise we could face numerous small issues.

Very good point to raise this here! We already have implemented a version of the algorithm that updates a single user's ratings when comparisons are submitted. @lenhoanglnh @amatissart what is the timeline and blockers to switch the video poll to Mehestan?